addons/l10n_uy/models/res_partner.py

```diff
@@ -0,0 +1,100 @@
+import logging
+import re
+
+from odoo import api, models, _
+
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.constrains('vat', 'l10n_latam_identification_type_id')
+    def check_vat(self):
+        # EXTEND account/base_vat
+        """ Add validation of UY document types CI and NIE """
+        ci_nie_types = self.filtered(
+            lambda p: p.l10n_latam_identification_type_id.l10n_uy_dgi_code in ('1', '3')
+            and p.l10n_latam_identification_type_id.country_id.code == 'UY' and p.vat)
+        for partner in ci_nie_types:
+            if not partner._l10n_uy_ci_nie_is_valid():
+                raise ValidationError(self._l10n_uy_build_vat_error_message(partner))
+        return super(ResPartner, self - ci_nie_types).check_vat()
+
+    @api.model
+    def _l10n_uy_build_vat_error_message(self, partner):
+        """ Similar to _build_vat_error_message but using latam doc type name instead of vat_label
+        NOTE: maybe can be implemented in master to l10n_latam_base for the use of different doc types """
+        vat_label = _('CI/NIE')
+        expected_format = '3:402.010-2 or 93:402.010-1 (CI or NIE)'
+        partner_label = _("partner [%s]", partner.name)
+        # Catch use case where the record label is about the public user (name: False)
+        if 'False' not in partner_label:
```

Comment:
Almost good, but imagine if the partner's name is "John False".
It's better to just check from `partner.name` instead
```suggestion
        if partner.name:
			  partner_label = _("partner [%s]", partner.name)